python/pyspark/sql/functions.py

```diff
@@ -108,12 +108,13 @@ def _invoke_binary_math_function(name: str, col1: Any, col2: Any) -> Column:
     Invokes binary JVM math function identified by name
     and wraps the result with :class:`~pyspark.sql.Column`.
     """
+
+    def align_type(c: Any) -> Union[str, Column]:
+        # For legacy reasons, the arguments here can be implicitly converted into column
+        return c if isinstance(c, (str, Column)) else lit(c)
```

Comment:
Unless we plan to reuse this in the future, I'd make more senses to just inline the expression.


If we do plan to reuse it, this should be marked as internal (`align_type` -> `_align_type`) and moved next to other helper functions (also, a docstring would be a good idea).